### ISSUE TITLE
CA2208 - False positive (in switch expersion?)

### ISSUE THREAD

**Issue by get-flat** [dotnet/roslyn-analyzers#7355]
### Analyzer

**Diagnostic ID**: [CA2208](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2208)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.303

### Steps To Reproduce

``` 
public void Method(Foo foo) 
{ 
    foo.FieldA = foo.FieldB switch
    {
        1 => "Bar",
        _ => throw new ArgumentOutOfRangeException(nameof(foo.FieldB), $"Invalid {foo.FieldB}"), // CA2208 !
    };
}
```

## Additional context
VS 17.10.4

**Comment by buyaa-n** [comment]
This is by design, it is warning because the parameter name is not exactly the method parameter 'foo', if would not warn if it was `throw new ArgumentOutOfRangeException(nameof(foo), $"Invalid {foo.FieldB}")`
